### PR TITLE
[improvement]Cached form.io token

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
Adding the Form.IO token to a cache to avoid requesting it multiple times while it is still valid.
The token will be stored locally and once it is expired or to about to expire (30s) it will be renewed.

### CODE REVIEW

File: sources/packages/api/src/services/form/form.service.ts [hunk 1]
```diff
@@ -119,14 +130,48 @@ export class FormService {
...
+  /**
+   * Tries get the token from the cache if it is not expired
+   * or about to be expired, otherwise get a new one.
+   * @returns Cached token or a new one.
+   */
+  private async tryGetCachedToken(): Promise<{ token: any; decoded: any }> {
+    const needRenewToken =
+      !this.cachedToken || // If the token is not present, retrieve it.
+      needRenewJwtToken(
+        +this.cachedToken.decoded["exp"],
```
Comment (ann-aot) [inline sources/packages/api/src/services/form/form.service.ts:159]: there is a `+` before this.cachedToken.decoded["exp"].

Comment (andrewsignori-aot) [reply sources/packages/api/src/services/form/form.service.ts:159]: This is to convert the string to a number in a "shortcut" way.
https://www.codegrepper.com/code-examples/typescript/how+to+convert+string+to+number+in+typescript

image

**Review by ann-aot — APPROVED** [review]
LGTM

**Review by guru-aot — APPROVED** [review]
LGTM

**Pull request merged**